experimental/oak_async/src/executor.rs

```diff
@@ -0,0 +1,176 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//! Async executor for Oak Nodes.
+
+use core::{cell::RefCell, future::Future, task::Waker};
+use futures::{executor::LocalPool, task::LocalSpawnExt};
+use log::{debug, trace};
+use oak::{ChannelReadStatus, Handle, OakStatus, ReadHandle};
+use std::collections::HashMap;
+
+// thread local so `cargo test` can run tests in parallel.
+// Oak nodes are single-threaded.
+std::thread_local! {
+    static EXECUTOR: RefCell<Executor> = RefCell::new(Executor::default());
+}
+
+type ReaderId = usize;
+
+struct WaitingReader {
+    handle: Handle,
+    waker: Waker,
+}
+
+/// Global executor context.
+///
+/// `with_executor` provides a way to get a handle to the global executor instance.
+#[derive(Default)]
+pub struct Executor {
```

Comment:
I believe this does not need to be `pub(crate)`, since the `executor` module that contains this is not public. Or is it convention to mark this as `pub(crate)` anyway?

Reply:
No convention, I just forgot to check the module visibility.